### PR TITLE
Fix esp32 getting started

### PR DESCRIPTION
**Pull request by szczys**
* Call out ESP-IDF in favor of ESP32
* Target tags when cloning repos
* Small code block syntax fix

### CODE REVIEW

File: docs/getting-started/3-device-examples/2-compile-example-code/3-esp-idf/README.md [hunk 1]
```diff
@@ -1,23 +1,22 @@
 ---
 id: overview
-title: ESP32 Overview
+title: ESP-IDF with Golioth
 ---
 
-## What is the ESP32?
+## What is the ESP-IDF?
```
Comment (sam-golioth) [inline docs/getting-started/3-device-examples/2-compile-example-code/3-esp-idf/README.md:6]:
```suggestion
## What is ESP-IDF?
```

File: docs/getting-started/3-device-examples/2-compile-example-code/3-esp-idf/README.md [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
+Systems](https://espressif.com/) for their line of ESP32 microcontrollers. The
+Golioth Firmware SDK can be included in your projects as an ESP-IDF component.
 
-The Espressif team has built-in support for Zephyr. This, in combination with the low cost and wide availability of the chips, has led to the ESP32 being one of the first platforms that Golioth has chosen to support.
+## Using ESP32 with Golioth
 
-## Terminology
+The ESP32 is a Wi-Fi and Bluetooth combination chip. There are multiple versions
+of the chip, with the most recent versions running in different configurations
+of core types, up to and including a RISC V core (on the ESP32-C3).
```
Comment (sam-golioth) [inline docs/getting-started/3-device-examples/2-compile-example-code/3-esp-idf/README.md:18]: "up to and including" sounds a little weird to me, like we're ranking architectures. This is existing copy though, so I'm fine with leaving this as is.

File: docs/partials-common/install-espidf.md [hunk 3]
```diff
@@ -26,10 +26,8 @@ sudo apt-get install git wget flex bison gperf python3 python3-venv \
 ```console
 mkdir -p ~/esp
 cd ~/esp
-git clone --recursive https://github.com/espressif/esp-idf.git
+git clone --recursive https://github.com/espressif/esp-idf.git -b v5.1.1
```
Comment (sam-golioth) [inline docs/partials-common/install-espidf.md:29]: Nice, much cleaner!

File: docs/partials-common/install-golioth-firmware-sdk.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ values={[
 
 ```console
 cd ~
-git clone --recursive https://github.com/golioth/golioth-firmware-sdk.git
+git clone https://github.com/golioth/golioth-firmware-sdk.git -b v0.9.0
```
Comment (sam-golioth) [inline docs/partials-common/install-golioth-firmware-sdk.md:19]: hmm why not `--recursive` here?

Comment (szczys) [reply docs/partials-common/install-golioth-firmware-sdk.md:19]: Two lines later we have them run `git submodule update --init --recursive` which seems redundant to me, but I can put it back in. There's no harm in having both as we do now.

File: docs/partials-common/install-espidf.md (hunk 3 above)
Comment (szczys) [reply docs/partials-common/install-espidf.md:29]: This is the recommended ESP-IDF approach and I agree it feels cleaner.

File: docs/getting-started/3-device-examples/2-compile-example-code/3-esp-idf/README.md (hunk 2 above)
Comment (szczys) [reply docs/getting-started/3-device-examples/2-compile-example-code/3-esp-idf/README.md:18]:
```
of the chip, including the ESP32, ESP32-S2, and ESP32-S3 based on the Xtensa
core and the ESP32-C3 which uses the RISC-V core.
```

You're right that this is wonky. I'll update to list parts and cores. It's a good place to indicate our breadth of support.

File: docs/partials-common/install-golioth-firmware-sdk.md (hunk 4 above)
Comment (sam-golioth) [reply docs/partials-common/install-golioth-firmware-sdk.md:19]: Can we just do `--recursive` without the `git submodule update` line? Like we do for downloading ESP-IDF.

Comment (szczys) [reply docs/partials-common/install-golioth-firmware-sdk.md:19]: Yeah, will do.

**Pull request merged**